modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/util/DDMFormValuesConverterUtil.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.util;
+
+import com.liferay.dynamic.data.mapping.form.field.type.constants.DDMFormFieldTypeConstants;
+import com.liferay.dynamic.data.mapping.model.DDMFormField;
+import com.liferay.dynamic.data.mapping.storage.DDMFormFieldValue;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Mateus Santana
+ */
+public class DDMFormValuesConverterUtil {
+
+	public static List<DDMFormFieldValue> addMissingDDMFormFieldValues(
+		Collection<DDMFormField> ddmFormFields,
+		Map<String, List<DDMFormFieldValue>> ddmFormFieldValues) {
+
+		List<DDMFormFieldValue> newDDMFormFieldValues = new ArrayList<>();
+
+		ddmFormFields.forEach(
+			ddmFormField -> {
+				List<DDMFormFieldValue> nestedDDMFormFieldValues =
+					new ArrayList<>();
+
+				if (StringUtil.equals(
+						ddmFormField.getType(),
+						DDMFormFieldTypeConstants.FIELDSET)) {
+
+					nestedDDMFormFieldValues.addAll(
+						addMissingDDMFormFieldValues(
+							ddmFormField.getNestedDDMFormFields(),
+							ddmFormFieldValues));
+				}
+
+				if (!ddmFormFieldValues.containsKey(ddmFormField.getName())) {
+					_addNewMissedDDMFormFieldValue(
+						ddmFormField, newDDMFormFieldValues,
+						nestedDDMFormFieldValues);
+				}
+				else {
+					if (ListUtil.isNotEmpty(nestedDDMFormFieldValues)) {
+						_addNewMissedDDMFormFieldValue(
+							ddmFormField, newDDMFormFieldValues,
+							nestedDDMFormFieldValues);
```

Comment:
@mateussgn, you weren't able to make a single call to the method `_addNewMissedDDMFormFieldValue`, right?

Did you prefer to keep the previous code instead of create a new variable to control the code flow, am I correct?

Reply:
I'm sorry, I ended up forgotten to add this change. But I've already done it today, and I'll update the branch.